src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy

```diff
@@ -71,7 +70,7 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     void write(List<? extends ProtoNode> nodeProtos) throws Exception {
         Session session = theSession()
         for (ProtoNode nodeProto : nodeProtos) {
-            writeToJcr(nodeProto, session)
+            new ProtoNodeDecorator(nodeProto).writeJcrNodesRecursively(session)
```

Comment:
Too much detail in the method name. From JcrNodesWriter's perspective (or any client), we only care that our ProtoNodeDecorator is written correctly as a JcrNodeDecorator. Whatever it takes to get to that point - details like writing mandatory child nodes recursively, and checking out versionables should be kept out of sight.